adr/adr-1.md

```diff
@@ -0,0 +1,124 @@
+
+The agent can execute a single JavaScript file (`.js`) or it can take an archive file as the artifact, where it will extract all `.js` files from the archive and run them in a single context, provided there remains only one exported `run` function.
+
+The JavaScript agent is responsible for exposing host services to each function. The agent must also ensure that no two workloads share the same execution context.
+
+### WebAssembly Agent
+The WebAssembly agent is responsible for executing WebAssembly _components_. We no longer support running the classic modules. All executable WebAssembly artifacts will be components with their interfaces defined using the `wit` convention (and likely the appropriate `wit-bindgen` tooling).
+
+The WebAssembly agent is responsible for exposing host services to each function, proxying access to the host services API exposed on the internal NATS server.
+
+### OCI Agent
+The OCI agent will be responsible for running images from an indicated container registry. This effectively gives Nex nodes the ability to schedule and start docker images.
+
+Information on how to connect to the appropriate host services URL/creds will be provided to the workload. OCI workloads are considered long-lived services and as such do not require a host services proxy.
+
+## New Agent Protocol
+Every agent is required to adhere to this agent protocol. The agents _must_ connect to the internal NATS server embedded in the Nex node host and this is the _only_ way that agents and Nex nodes communicate with each other other than when the Nex node initially spawns an agent.
+
+Note that in the table below, _direction_ is relative to the _agent_. `hostint.>` is an export from the Nex node account, while `agentint.>` is an export from the agent's user/account. All communications in the table below happens on the internal connection, _agents should not communicate with any NATS servers other than the Nex node host_.
+
+| Subject | Direction | Description |
+|--|--|:--|
+| `hostint.{workload-type}.register` | **request** | Registers the agent information with the Nex node host. Here `workload-type` is the name of the workload type this agent handles, e.g. `native` or `javascript` |
+| `hostint.{workload-type}.logs` | **publish** | Emits logs from running workloads for the Nex node to capture and potentially re-broadcast |
+| `hostint.{workload-type}.events.{event-type}` | **publish** | Emits a cloud event from the agent to the Nex node |
+| `hostint.hostservices.<service>.<method>` | **request** | When proxying host services calls, the agent will use this subject. The calling workload will be indicated with `x-agent-workloadid`, `x-agent-workloadtype`, and `x-agent-namespace` headers. The raw payload is passed directly. For example, a JavaScript function making a key-value `get` request would make a request on `hostint.hostservices.keyvalue.get`. Additional headers for e.g. metrics/tracing or for specific host services can also be passed. |
```

Comment:
Curious the motivation for not putting these header values as tokens in the subject?